### PR TITLE
[DEV-1864] Refactor FeatureManager and TileManager as services

### PR DESCRIPTION
**Pull request by YS-L**
## Description

This refactors `FeatureManager` and `TileManager` as services that inherit from `BaseService`. This makes it easier for us to start using mongo to keep track of states such as tile registry, tile feature mapping, online store mapping etc that are currently tracked in data warehouse unnecessarily.

Also added `HistoricalFeaturesService` and `TileCacheService` which depend on the two services above. This keeps the import dependencies clean and allows us to move logic with side effects out of `featurebyte/query_graph/sql/feature_historical.py` which ideally should only concern sql generation.

There is no functional changes: all the changes are refactoring and moving codes around.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/service/online_enable.py [hunk 1]
```diff
@@ -6,9 +6,7 @@
 from typing import Any, Optional
 
 from bson.objectid import ObjectId
-from pydantic import PrivateAttr
 
-from featurebyte.feature_manager.manager import FeatureManager
 from featurebyte.feature_manager.model import ExtendedFeatureModel
```
Comment (gee-senbong) [inline featurebyte/service/online_enable.py:10]: 👍 Good to move `FeatureManager` to service directory. Would be good if we can also move the `ExtendedFeatureModel` from `featurebyte.feature_manager.model` to `featurebyte.models` (definitely should do it in another PR).

File: featurebyte/service/preview.py [hunk 2]
```diff
@@ -445,82 +436,6 @@ async def preview_featurelist(
...
-            graph=feature_cluster.graph,
-            nodes=feature_cluster.nodes,
-            observation_set=observation_set,
-            serving_names_mapping=featurelist_get_historical_features.serving_names_mapping,
-            source_type=feature_store.type,
-            is_feature_list_deployed=is_feature_list_deployed,
-            parent_serving_preparation=parent_serving_preparation,
-            output_table_details=output_table_details,
-            progress_callback=progress_callback,
-        )
```
Comment (gee-senbong) [inline featurebyte/service/preview.py:522]: 👍 Good to move this to `historical_feature_service`.

File: tests/integration/conftest.py [hunk 3]
```diff
@@ -960,12 +963,12 @@ def snowflake_feature_expected_tile_spec_dict_fixture():
...
-    """
-    Feature Manager fixture
-    """
-    return FeatureManager(session=session)
+# @pytest.fixture
+# def feature_manager(session):
+#     """
+#     Feature Manager fixture
+#     """
+#     return FeatureManager(session=session)
```
Comment (gee-senbong) [inline tests/integration/conftest.py:971]: Should we remove this? Or this is intended?

**Review by gee-senbong — APPROVED** [review]
LGTM. Nice refactoring work!

File: tests/integration/conftest.py (hunk 3 above)
Comment (YS-L) [reply tests/integration/conftest.py:971]: It should be removed. Will fix.

File: featurebyte/service/tile_cache.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+from featurebyte.logging import get_logger
+from featurebyte.persistent import Persistent
+from featurebyte.query_graph.graph import QueryGraph
+from featurebyte.query_graph.node import Node
+from featurebyte.service.base_service import BaseService
+from featurebyte.service.tile_manager import TileManagerService
+from featurebyte.session.base import BaseSession
+from featurebyte.tile.tile_cache import TileCache
+
+logger = get_logger(__name__)
```
Comment (jevonyeoh) [inline featurebyte/service/tile_cache.py:19]: looks unused

File: featurebyte/service/tile_manager.py [hunk 5]
```diff
@@ -403,12 +413,12 @@ async def remove_tile_jobs(
         TileScheduleNotSupportedError
             if task manager is not initialized
         """
-        if not self._task_manager:
+        if not self.task_manager:
```
Comment (jevonyeoh) [inline featurebyte/service/tile_manager.py:416]: not really related to this PR, but i wonder if some of these dependency checks can be simplified with by injecting them in now too

**Review by jevonyeoh — APPROVED** [review]

**Review by yanhuisg — APPROVED** [review]
LGTM

File: featurebyte/service/tile_manager.py (hunk 5 above)
Comment (yanhuisg) [reply featurebyte/service/tile_manager.py:416]: yes. would be nice to make use of app_container or sth similar as well

Comment (YS-L) [reply featurebyte/service/tile_manager.py:416]: This is already using dependency injection so this check should be redundant now. Will remove in a next PR.
